app/models/status.rb

```diff
@@ -440,6 +440,12 @@ def self._compile_reblog_insert(values)
     im
   end
 
+  def discard_with_reblogs
+    discard_time = Time.current
+    Status.unscoped.where(reblog_of_id: id, deleted_at: [nil, deleted_at]).in_batches.update_all(deleted_at: discard_time) unless reblog?
```

Comment:
It might be good to wrap this in a transaction, perhaps?

Reply:
I am not sure why, I don't think any error can occur here, and another process observing an intermediate step would not be an issue since reblogs are discarded before the reblogged status.